integration/indexes/materialized_view_test.rb

```diff
@@ -0,0 +1,323 @@
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#++
+
+require File.dirname(__FILE__) + '/../integration_test_case.rb'
+
+class MaterializedViewTest < IntegrationTestCase
+
+  def setup
+    return if CCM.cassandra_version < '3.0.0'
+
+    @@ccm_cluster.setup_schema("CREATE KEYSPACE simplex WITH replication = {'class': 'SimpleStrategy', 'replication_factor': 1}")
+
+    @cluster = Cassandra.cluster(
+        schema_refresh_delay: 0.1,
+        schema_refresh_timeout: 0.1
+    )
+    @listener = SchemaChangeListener.new(@cluster)
+    @session = @cluster.connect('simplex')
+
+    @session.execute("CREATE TABLE simplex.scores(
+                        user TEXT,
+                        game TEXT,
+                        year INT,
+                        month INT,
+                        day INT,
+                        score INT,
+                        PRIMARY KEY (user, game, year, month, day)
+                        )")
+    @session.execute("CREATE MATERIALIZED VIEW simplex.monthlyhigh AS
+                        SELECT game, year, month, score, user, day FROM simplex.scores
+                        WHERE game IS NOT NULL AND year IS NOT NULL AND month IS NOT NULL AND score IS NOT NULL AND user IS NOT NULL AND day IS NOT NULL
+                        PRIMARY KEY ((game, year, month), score, user, day)
+                        WITH CLUSTERING ORDER BY (score DESC, user ASC, day ASC)")
+    @listener.wait_for_materialized_view('simplex', 'monthlyhigh')
+  end
+
+  def teardown
+    @cluster && @cluster.close
+  end
+
+  # Test for retrieving materialized view metadata
+  #
+  # test_can_retrieve_materialized_view_metadata tests that all pieces of materialized view metadata can be retrieved.
+  # It goes through each piece of materialized view metadata and verifies that each piece is as expected.
+  #
+  # @since 3.0.0
+  # @jira_ticket RUBY-167
+  # @expected_result materialized view metadata should be retrieved.
+  #
+  # @test_category materialized_view
+  #
+  def test_can_retrieve_materialized_view_metadata
+    skip("Materialized views were introduced in Cassandra 3.0.0") if CCM.cassandra_version < '3.0.0'
+
+    assert @cluster.keyspace('simplex').has_materialized_view?('monthlyhigh')
+    mv_meta = @cluster.keyspace('simplex').materialized_view('monthlyhigh')
+
+    assert_equal 'monthlyhigh', mv_meta.name
+    refute_nil mv_meta.id
+    assert_equal 'scores', mv_meta.base_table.name
+    assert_equal 'simplex', mv_meta.keyspace.name
+    refute_nil mv_meta.options
+
+    assert_equal 6, mv_meta.primary_key.size
```

Comment:
This gets kind of verbose and repetitive. How about defining an assert_columns method and calling it once per column-set, something like this:
```
def assert_columns(expected_names_and_types, actual_columns)
  assert_equal(expected_names_and_types.size, actual_columns.size)
  expected_names_and_types.zip(actual_columns) do |expected, actual_column|
    assert_equal expected[0], actual_column.name
    assert_equal expected[1], actual_column.type.kind
  end
end
```
and then in a call site, do something like this:
```assert_columns([['game', :text], ['year', :int], ['month', :int], ['score', :int], ['user', :text], ['day', :int]], mv_meta.primary_key)```

If the nested array bugs you, flatten it and change assert_columns accordingly; expected_names_and_columns.size == 2 * actual_columns.size, and you'd have to implement the loop differently...or unflatten the array in assert_columns.